File: Correctness-and-the-Loop-Invariant.cs
using System;
using System.Linq;

class Solution
{
    static void InsertionSort2(int n, int[] arr)
    {
        for (int i = 1; i < n; ++i)
        {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key)
            {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }

        ShowArray(arr);
    }

    static void ShowArray(int[] arr)
    {
        Console.WriteLine(string.Join(" ", arr));
    }

    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        int[] arr = Console.ReadLine()
                           .Trim()
                           .Split(' ')
                           .Select(int.Parse)
                           .ToArray();

        InsertionSort2(n, arr);
    }
}

File: Correctness-and-the-Loop-Invariant.java
import java.util.*;

public class Solution {

    public static void insertionSort2(int n, int[] arr) {
        for (int i = 1; i < n; ++i) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }

        showArray(arr);
    }

    public static void showArray(int[] arr) {
        for (int num : arr)
            System.out.print(num + " ");
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        String[] arrStr = scanner.nextLine().trim().split("\\s+");

        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = Integer.parseInt(arrStr[i]);

        insertionSort2(n, arr);
    }
}
